src/main/java/uk/gov/hmcts/reform/sscs/helper/service/HearingsServiceHelper.java

```diff
@@ -1,34 +1,77 @@
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
 import uk.gov.hmcts.reform.sscs.model.HearingEvent;
 import uk.gov.hmcts.reform.sscs.model.HearingWrapper;
 import uk.gov.hmcts.reform.sscs.model.single.hearing.HmcUpdateResponse;
 
-import java.util.Optional;
+import java.util.ArrayList;
+import javax.validation.Valid;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
 
 public final class HearingsServiceHelper {
 
     private HearingsServiceHelper() {
     }
 
-    public static void updateHearingId(HearingWrapper wrapper, HmcUpdateResponse response) {
-        wrapper.getCaseData().getSchedulingAndListingFields().setActiveHearingId(response.getHearingRequestId());
+    public static void updateHearingId(Hearing hearing, HmcUpdateResponse response) {
+        if (nonNull(response.getHearingRequestId())) {
+            hearing.getValue().setHearingId(String.valueOf(response.getHearingRequestId()));
+        }
     }
 
-    public static void updateVersionNumber(HearingWrapper wrapper, HmcUpdateResponse response) {
-        wrapper.getCaseData().getSchedulingAndListingFields().setActiveHearingVersionNumber(response.getVersionNumber());
+    public static void updateVersionNumber(Hearing hearing, HmcUpdateResponse response) {
+        hearing.getValue().setVersionNumber(response.getVersionNumber());
     }
 
     public static HearingEvent getHearingEvent(HearingState state) {
         return HearingEvent.valueOf(state.name());
     }
 
     public static String getHearingId(HearingWrapper wrapper) {
-        return Optional.of(wrapper)
-            .map(HearingWrapper::getCaseData)
-            .map(SscsCaseData::getSchedulingAndListingFields)
-            .map(SchedulingAndListingFields::getActiveHearingId)
-            .map(Object::toString).orElse(null);
+        Hearing hearing = wrapper.getCaseData().getLatestHearing();
+        if (nonNull(hearing)) {
+            return hearing.getValue().getHearingId();
+        }
+        return null;
+    }
+
+    public static Long getVersion(HearingWrapper wrapper) {
+        Hearing hearing = wrapper.getCaseData().getLatestHearing();
+        if (nonNull(hearing)) {
+            Long version = hearing.getValue().getVersionNumber();
+            if (nonNull(version) && version > 0) {
+                return version;
+            }
+        }
+        return null;
+    }
+
+    public static Hearing createHearing(Long hearingId, @Valid SscsCaseData caseData) {
+        Hearing hearing = Hearing.builder()
+            .value(HearingDetails.builder()
+                .hearingId(String.valueOf(hearingId))
+                .build())
+            .build();
+        caseData.getHearings().add(hearing);
+        return hearing;
+    }
+
+    @Nullable
+    public static Hearing getHearingById(Long hearingId, @Valid SscsCaseData caseData) {
+        if (isNull(caseData.getHearings())) {
+            caseData.setHearings(new ArrayList<>());
```

Comment:
minor/optional - preferably we should use Collections.emptyList() or List.of() instead of ArrayList - this practice is derived from Liskov Substitution Principle of the SOLID principles.

Reply:
Same again:
![image](https://user-images.githubusercontent.com/18538046/173393296-0d1df5e2-3bb1-402f-9d3d-44ea7a2d86a2.png)